File: cplusplus/design_pattern/DecoratorProj/ConcreteDecorator.cpp
#include "stdafx.h"
#include "ConcreteDecorator.h"


ConcreteDecorator::ConcreteDecorator()
{
}

ConcreteDecorator::ConcreteDecorator(std::shared_ptr<IComponent> component):Decorator(component)
{
}


ConcreteDecorator::~ConcreteDecorator()
{
}

void ConcreteDecorator::showName()
{
    Decorator::showName();
    std::cout << "ConcreteDecorator::showName()" << std::endl;
}

File: Java/pattern/Decorator/src/com/pattern/ConcreteDecorator.java
package com.pattern;

public class ConcreteDecorator extends Decorator {

	public ConcreteDecorator(IComponent m_component) {
		super(m_component);
	}
	
	public void showName() {
		super.showName();
		System.out.println("ConcreteDecorator");
	}

}
